src/Faithlife.WebRequests/Json/ExplicitStatusResponse.cs

```diff
@@ -0,0 +1,32 @@
+using System.Net;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Faithlife.WebRequests.Json
+{
+	/// <summary>
+	/// AutoWebServiceRequest responses with an explicitly set status code. 
+	/// </summary>
+	public abstract class ExplicitStatusResponse : AutoWebServiceResponse
```

Comment:
What's the rationale for exposing this as a public type from `Faithlife.WebRequests.Json`? I'm not immediately understanding why user code doesn't just add a `StatusCode` property to their response object (since they have to implement a derived type anyway).

Reply:
I agree; this class should be removed. If clients want a common base class for including these properties, they can create it themselves.

Reply:
This base class is useful for writing a library of client aspects that can operate on any response type. The aspect necessary for the current use-case is for an AuthenticationLostHandler that mirrors [this WebCommon code](https://git/Logos/WebCommon/blob/nuget/src/Logos.Common.Authentication/AuthenticationLostUtility.cs#L7-L12).

We could put this base-class inside whatever library wants to contain the stable of extensions, but then what if I've got two libraries I want to use? C# doesn't do multiple-inheritance well, so now I've probably got an interface for each library, and I need to know at client-package-publish time which libraries my consumer will want to use.

At the point where this class adds very little burden on `Faithlife.WebRequests`, and enhances it in some ways by making clear a use-case for the `StatusCode` magic, I see enough benefit to keep it around.

Do you have any objections other than the mentioned YAGNI?

Reply:
That's fair. The name throws me a bit, though. `AnyStatusCodeResponse`?

Reply:
@tobs1701, what do you think of that name?

Reply:
I think I would prefer something like `GenericStatusCodeResponse`, though `AnyStatusCodeResponse` works in a pinch.

Reply:
I have no objection to `GenericStatusCodeResponse`.

Reply:
Also, we might consider removing `abstract`. The class is potentially useful on its own.

Reply:
Retained per discussion, sans `abstract` and with an updated name. 9c6a8a47ed60beacfa362fcb6831fa078b832668